src/include/execution/sql/decimal_magic_numbers.h

```diff
@@ -0,0 +1,286 @@
+#pragma once
+
+#include <unordered_map>
+
+namespace noisepage::execution::sql {
+
+/** The following hash is required to create an unordered map of 128bit uints*/
+struct Unsigned128BitHash {
+  /** Hashing function
+   * @param value - the value to be hashed*/
+  hash_t operator()(const uint128_t &value) const {
+    uint128_t x = value;
+    const uint64_t k_mul = 0x9ddfea08eb382d69ULL;
```

Comment:
Can you document the source of this? If Hacker's Delight, can you provide the section?

Reply:
You can do this at the top of the file

Reply:
it is mentioned in the PR